packets.go

```diff
@@ -422,6 +440,39 @@ func (mc *mysqlConn) writeNativeAuthPacket(cipher []byte) error {
 	return mc.writePacket(data)
 }
 
+//  Caching sha2 authentication. Public key request and send encrypted password
+// http://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::AuthSwitchResponse
+func (mc *mysqlConn) writePublicKeyAuthPacket(cipher []byte) error {
+	// request public key
+	data := mc.buf.takeSmallBuffer(4 + 1)
+	data[4] = cachingSha2PasswordRequestPublicKey
+	mc.writePacket(data)
+
+	data, err := mc.readPacket()
+	if err != nil {
+		return err
+	}
+
+	block, _ := pem.Decode(data[1:])
+	pub, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return err
+	}
+
+	plain := make([]byte, 20)
+	for k, v := range []byte(mc.cfg.Passwd) {
+		plain[k] = byte(v)
+	}
+	for i := range plain {
+		plain[i] ^= cipher[i]
```

Comment:
`len(plain) == 20`, but how about `len(cipher)`?
Shouldn't we do `cipher[i % len(cipher)]`?

Reply:
It's cipher is Nonce  https://dev.mysql.com/doc/dev/mysql-server/latest/page_caching_sha2_authentication_exchanges.html
from server and always 20 bytes

Reply:
I see.  Please fix only above. (`len(mc.cfg.Passwd) > 20` case).

Reply:
Sorry, There is not password length limitation (< 20)
Now fix it 84f6018